components/ContactTimeline/src/ContactTimeline.tsx

```diff
@@ -0,0 +1,192 @@
+import { ContactTimelineMetaItem } from './ContactTimelineMetaItem';
+import { ContactTimelineMetaTimeItem } from './ContactTimelineMetaTimeItem';
+import { ContactTimelineMeta } from './ContactTimelineMeta';
+import { ContactTimelineHeaderContent } from './ContactTimelineHeaderContent';
+import { ContactTimelineHeaderDate } from './ContactTimelineHeaderDate';
+import { ContactTimelineHeaderChannel } from './ContactTimelineHeaderChannel';
+import { Paragraph } from '@gemeente-denhaag/typography';
+import { ContactTimelineItemSender } from './ContactTimelineItemSender';
+import { ContactTimelineItemFile } from './ContactTimelineItemFile';
+
+export interface ContactTimelineItemProps {
+  id: Key;
+  title: ReactNode;
+  description?: ReactNode;
+  file?: ReactNode;
+  sender?: ReactNode;
+  date?: ReactNode;
+  isoDate?: string;
+  channel: ReactNode;
+}
+
+interface ContactTimelineItemInternalProps extends ContactTimelineItemProps {
+  todayLabel: string;
+  nextItem?: boolean;
+  expanded?: boolean;
+  toggleExpanded: false | (() => void);
+}
+
+export interface ContactTimelineProps {
+  items: ContactTimelineItemProps[];
+  expandedItems?: Key[];
+  collapsible?: boolean;
+  todayLabel: string;
+}
+
+const toggleState = (key: Key, collection: Key[], setCollection: React.Dispatch<React.SetStateAction<React.Key[]>>) => {
+  if (collection.includes(key)) {
+    setCollection(collection.filter((item) => item !== key));
+  } else {
+    setCollection([...collection, key]);
+  }
+};
+
+export const ContactTimeline: React.FC<ContactTimelineProps> = ({
+  items,
+  todayLabel,
+  expandedItems: initialExpanded = [],
+  collapsible = false,
+}: ContactTimelineProps) => {
+  const [expandedItems, setexpandedItems] = useState(initialExpanded);
+
+  return (
+    <ContactTimelineList className="denhaag-process-steps">
+      {items.map((item, index) => {
+        const nextItem = index < items.length - 1;
+        return (
+          <ContactTimelineListItem
+            {...item}
+            key={item.id}
+            expanded={collapsible ? expandedItems.includes(item.id) : false}
+            nextItem={nextItem}
+            todayLabel={todayLabel}
+            toggleExpanded={
+              collapsible &&
+              (() => {
+                toggleState(item.id, expandedItems, setexpandedItems);
+              })
+            }
+          />
+        );
+      })}
+    </ContactTimelineList>
+  );
+};
+
+export interface ContactTimelineListProps extends OlHTMLAttributes<HTMLOListElement> {}
+
+export const ContactTimelineList: React.FC<ContactTimelineListProps> = ({
+  className,
+  children,
+  ...props
+}: ContactTimelineListProps) => (
+  <ol {...props} className={clsx('denhaag-contact-timeline', className)}>
+    {children}
+  </ol>
+);
+
+interface Props {
+  dateTime: string;
+  todayLabel: string;
+  now?: string;
+  locale?: Locale;
+}
+
+export const formatContactTimelineDate = ({
```

Comment:
Kunnen we dit verplaatsen naar een Global function?

Reply:
Ja, denk dat we dit wat generieker en global moeten maken, samen met de format date van o.a. het Action component. Zodat wel altijd dezelfde datum formatter + logica van vandaag/gisteren/eergisteren hebben.